packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/globals.dart';
+import 'package:json_schema/json_schema.dart';
+import 'package:yaml/yaml.dart';
+
+import 'base/file_system.dart';
+import 'cache.dart';
+
+/// A wrapper around the `flutter` section in the  `pubspec.yaml` file.
+class FlutterManifest {
+  Map<String, dynamic> _descriptor;
+  Map<String, dynamic> _flutterDescriptor;
+
+  FlutterManifest._();
+
+  String get appName => _descriptor['name'];
+
+  bool get usesMaterialDesign {
+    return _flutterDescriptor.containsKey('uses-material-design') &&
+        _flutterDescriptor['uses-material-design'];
+  }
+
+  List<Map<String, dynamic>> get fontsDescriptor {
+    if (!_flutterDescriptor.containsKey('fonts')) {
+      return <Map<String, dynamic>>[];
+    }
+    return _flutterDescriptor['fonts'];
+  }
+
+  List<String> get assets {
+    if (!_flutterDescriptor.containsKey('assets')) {
+      return <String>[];
+    }
+    return _flutterDescriptor['assets'];
+  }
+
+  List<Font> _fonts;
+
+  List<Font> get fonts {
+    if (!_flutterDescriptor.containsKey('fonts')) {
+      return <Font>[];
+    }
+    if (_fonts == null) {
+      final List<Font> fonts = <Font>[];
+      for (Map<String, dynamic> fontFamily in _flutterDescriptor['fonts']) {
+        final List<Map<String, dynamic>> fontFiles = fontFamily['fonts'];
+        final String familyName = fontFamily['family'];
+        if (familyName == null) {
+          printError('Warning: Missing family name for font.', emphasis: true);
+          continue;
+        }
+        if (fontFiles == null) {
+          printError('Warning: No fonts specified for font $familyName', emphasis: true);
+          continue;
+        }
+
+        final List<FontAsset> fontAssets = <FontAsset>[];
+        for (Map<String, dynamic> fontFile in fontFiles) {
+          final String asset = fontFile['asset'];
+          if (asset == null) {
+            printError('Warning: Missing asset in fonts for $familyName', emphasis: true);
+            continue;
+          }
+
+          fontAssets.add(new FontAsset(
+            asset,
+            weight: fontFile['weight'],
+            style: fontFile['style'],
+          ));
+        }
+        if (fontAssets.isNotEmpty)
+          fonts.add(new Font(fontFamily['family'], fontAssets));
+      }
+
+     _fonts = fonts;
+    }
+
+    return _fonts;
+  }
+
+  static Future<FlutterManifest> createManifestFromPath(String path) async {
```

Comment:
Document that these will return `null` if the manifest was unable to be created because of an invalid format, missing file, etc.